### PR TITLE
kevinsolorio/tg-14-create-the-choose-your-state-and-plot-status-pages

### PR DESCRIPTION
**Pull request by kevin3656**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

added the plot and state pages for onboarding, a profile management, and style folder with colors

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
check out new onboarding folder and the pgaes/styling. Run app and go to /onboarding/plotstatus or /onboarding/state to checkout

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Working on the profile context still, for this pr I just made an array that should store a profile information.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
Next PR should expand on this one

CC: @ccatherinetan

### CODE REVIEW

File: app/onboarding/plotstatus/page.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    addProfile(newProfile);
+    console.log('test');
+  };
+  return (
+    <PageContainer>
+      <ContentContainer>
+        <h1>Plot Status</h1>
+        <h3>Do you own a plot to plant in?</h3>
+        <ButtonContainer>
+          <Button onClick={() => handleButtonClick('Yes')}>Georgia</Button>
```
Comment (rachaelch3n) [inline app/onboarding/plotstatus/page.tsx:25]: I think Yes and Georgia are flipped. Should be ('Georgia')}>Yes

File: app/onboarding/plotstatus/page.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+'use client';
```
Comment (ccatherinetan) [inline app/onboarding/plotstatus/page.tsx:1]: let's delete the `app/onboarding/page.tsx` file!

File: types/schema.d.ts [hunk 3]
```diff
@@ -2,6 +2,13 @@ import type { UUID } from 'crypto';
 
 export type Season = 'SPRING' | 'SUMMER' | 'FALL' | 'WINTER';
 
+export interface Profile {
+  state: string;
+  email: string;
+  phone_num: string;
```
Comment (ccatherinetan) [inline types/schema.d.ts:8]: quick note: we may remove phone_num in the future (gonna discuss w rob on fri), but we can leave it for now.

File: types/schema.d.ts [hunk 4]
```diff
@@ -2,6 +2,13 @@ import type { UUID } from 'crypto';
 
 export type Season = 'SPRING' | 'SUMMER' | 'FALL' | 'WINTER';
 
+export interface Profile {
```
Comment (ccatherinetan) [inline types/schema.d.ts:5]: we also want to include `user_id: UUID` 
let's reference the [notion](https://www.notion.so/calblueprint/Typescript-Schema-942a881ff45f408aa0b977d5c3859a5f) as the source of truth for schema!

File: app/onboarding/state/page.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  ButtonContainer,
+  ContentContainer,
+  PageContainer,
+} from '../styles';
+
+export default function Page() {
+  const { addProfile } = useProfile();
+
+  const handleButtonClick = (state: string) => {
+    const newProfile = { state, email: '', phone_num: '', user_type: '' };
```
Comment (ccatherinetan) [inline app/onboarding/state/page.tsx:15]: it might be cleaner to type this as Partial<Profile>, i.e.

`const newProfile: Partial<Profile> = { state };`
For more info on the Partial type: https://dev.to/martinpersson/better-code-quality-with-typescripts-utility-types-pick-partial-and-omit-3605

the creation of a new profile should be done outside the handleButtonClick function, since we want to udpate the same profile on each button click

File: hooks/useProfile.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+import { Profile } from '../types/schema';
+
+const initialProfiles: Profile[] = [];
```
Comment (ccatherinetan) [inline hooks/useProfile.ts:4]: ooo this is interesting for testing! we would ultimately replace this with pushing the profiles directly to Supabase though. 
if we're including it here, do you want to actually use it to test the functionality of your components?
in either case, before we merge this, we probably want to remove or comment out code used for testing

File: app/onboarding/plotstatus/page.tsx (hunk 1 above)
Comment (ccatherinetan) [reply app/onboarding/plotstatus/page.tsx:25]: good point! I think it should be 
`<Button onClick={() => handleButtonClick('Yes')}>Yes</Button>` and the line below:
`<Button onClick={() => handleButtonClick('No')}>No</Button>`
Perhaps one way to reduce these types of errors is to create a list of possible options and map them into button components, so there's 1 source of truth for each button option

File: app/onboarding/plotstatus/page.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  Button,
+  ButtonContainer,
+  ContentContainer,
+  PageContainer,
+} from '../styles';
+
+export default function Page() {
+  const { addProfile } = useProfile();
+
+  const handleButtonClick = (state: string) => {
```
Comment (ccatherinetan) [inline app/onboarding/plotstatus/page.tsx:14]: let's make sure to update this to handle the plot status! 
and use the `Partial<Profile>` type for the line below

**Review by ccatherinetan — COMMENTED** [review]
looking good so far! please see my comments; some of the comments on the plotstatus page apply to the state page and vice versa. 
1. use `Partial<Profile>` to define and update profiles 
2. for now, define a new profile at the top of each component, instead of inside the handleButtonClick function
3. consider mapping the button options so that there's a single source of truth

File: app/onboarding/state/page.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    addProfile(newProfile);
+    console.log('test');
```
Comment (ccatherinetan) [inline app/onboarding/state/page.tsx:17]: Maybe we can `console.log(newProfile)` instead!
